tests/unit/src/test/scala/tests/decorations/SyntheticHoverLspSuite.scala

```diff
@@ -0,0 +1,64 @@
+package tests.decorations
+
+import tests.BaseLspSuite
+
+class SyntheticHoverLspSuite extends BaseLspSuite("implicits") {
+
+  test("hovers") {
+    for {
+      _ <- server.initialize(
+        s"""
+           |/metals.json
+           |{
+           |  "a": {}
+           |}
+           |
+           |/a/src/main/scala/com/example/Main.scala
+           |package com.example
+           |import scala.concurrent.Future
+           |case class Location(city: String)
+           |object Main{
+           |  def hello()(implicit name: String, from: Location) = {
+           |    println(s"Hello $$name from $${from.city}")
+           |  }
+           |  implicit val andy : String = "Andy"
+           |
+           |  def greeting() = {
+           |    implicit val boston = Location("Boston")
+           |    hello()
+           |  }
+           |  
+           |  "foo".map(c => c.toUpper)
+           |}
+           |""".stripMargin
+      )
+      _ <- server.didChangeConfiguration(
+        """{
+          |  "implicit-argument-annotations-enabled": true,
+          |  "type-annotations-enabled": true
+          |}
+          |""".stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/com/example/Main.scala")
+      _ <- server.didSave("a/src/main/scala/com/example/Main.scala")(identity)
+      _ <- server.assertHoverAtLine(
+        "a/src/main/scala/com/example/Main.scala",
+        "    hello()@@",
+        """|**This line with synthetics added**:
+           |```scala
+           |hello()(com.example.Main.andy, boston)
+           |```
+           |""".stripMargin
+      )
+      _ <- server.assertHoverAtLine(
+        "a/src/main/scala/com/example/Main.scala",
+        "  \"foo\".map(c @@=> c.toUpper)",
+        """|**This line with synthetics added**:
```

Comment:
@ckipp01 Change diagnostics to displaying the entire line with hover with shown additional synthetic info. What do you think?

Reply:
So playing around with this, for me personally, this is _WAY_ nicer. I really like it. The only tricky part is that the user isn't notified that this extra information is here. However, like the original screen shot, if they are notified with the info diagnostic, imo it's way way too noisy. When playing around with it again before you added hovers it bummed me out because I really wanted to like it, but just having the info in randomly in the hover with no context wasn't super helpful. It was also super distracting to see the info diagnostics literally everywhere.

I think this solution is a much better approach 👍

Reply:
<img width="643" alt="Screenshot 2020-10-01 at 13 34 36" src="https://user-images.githubusercontent.com/13974112/94804295-e094a280-03ea-11eb-8282-2a783446e7c8.png">

Here is what that example looks like in real life if you're curious.

Reply:
Yeah, diagnostics are way to noisy and could be quite fragile.

I wonder why the markdown is not working there, seemed to work ok in VS Code. Any idea?

Reply:
And a larger example:

<img width="761" alt="Screenshot 2020-10-01 at 13 39 35" src="https://user-images.githubusercontent.com/13974112/94804711-92cc6a00-03eb-11eb-8fdc-820d299667d5.png">

Reply:
> I wonder why the markdown is not working there, seemed to work ok in VS Code. Any idea?

I answered down below with more info, but basically I'm in a terminal ha. So there is no markdown rendering.